openmc/data/resonance_covariance.py

```diff
@@ -0,0 +1,812 @@
+                mean = mean_array.flatten()
+                par_samples = np.random.multivariate_normal(mean, cov,
+                                                            size=n_samples)
+                for sample in par_samples:
+                    energy = sample[0::3]
+                    gn = sample[1::3]
+                    gg = sample[2::3]
+                    records = []
+                    for j, E in enumerate(energy):
+                        records.append([energy[j], l_value[j], spin[j], gn[j],
+                                        gg[j], gfa[j], gfb[j]])
+                    columns = ['energy', 'L', 'J', 'neutronWidth',
+                               'captureWidth', 'fissionWidthA', 'fissionWidthB']
+                    sample_params = pd.DataFrame.from_records(records, columns=columns)
+                    res_range = copy.copy(resonances)
+                    res_range._prepared = False # Set prepared to False to ensure
+                                                # the sampled parameters are used
+                                                # in reconstruction
+                    res_range.parameters = sample_params
+                    samples.append(res_range)
+    
+            elif mpar == 5:
+                param_list = ['energy', 'neutronWidth', 'captureWidth',
+                              'fissionWidthA', 'fissionWidthB']
+                mean_array = pd.DataFrame.as_matrix(parameters[param_list])
+                spin = pd.DataFrame.as_matrix(parameters['J'])
+                l_value = pd.DataFrame.as_matrix(parameters['L'])
+                mean = mean_array.flatten()
+                par_samples = np.random.multivariate_normal(mean, cov,
+                                                            size=n_samples)
+                for sample in par_samples:
+                    energy = sample[0::5]
+                    gn = sample[1::5]
+                    gg = sample[2::5]
+                    gfa = sample[3::5]
+                    gfb = sample[4::5]
+                    records = []
+                    for j, E in enumerate(energy):
+                        records.append([energy[j], l_value[j], spin[j], gn[j],
+                                        gg[j], gfa[j], gfb[j]])
+                    columns = ['energy', 'L', 'J', 'neutronWidth',
+                               'captureWidth', 'fissionWidthA', 'fissionWidthB']
+                    sample_params = pd.DataFrame.from_records(records, columns=columns)
+                    res_range = copy.copy(resonances)
+                    res_range._prepared = False # Set prepared to False to ensure
+                                                # the sampled parameters are used
+                                                # in reconstruction
+                    res_range.parameters = sample_params
+                    samples.append(res_range)
+    
+        self.samples = samples
```

Comment:
As we discussed before, the `samples` object should be returned directly rather than being set as an attribute of the current instance.